File: ios/GitaVani/GitaVani/Models/Translation.swift
import Foundation

struct Translation: Codable, Identifiable {
    let author: String
    let language: Language
    let text: String

    var id: String { "\(author)-\(language.rawValue)" }
}

struct Commentary: Codable, Identifiable {
    let author: String
    let language: Language
    let text: String

    var id: String { "\(author)-\(language.rawValue)-commentary" }
}

File: android/GitaVani/app/src/main/java/com/nikhilsi/gitavani/model/Translation.kt
package com.nikhilsi.gitavani.model

import kotlinx.serialization.Serializable

@Serializable
data class Translation(
    val author: String,
    val language: Language,
    val text: String
)

@Serializable
data class Commentary(
    val author: String,
    val language: Language,
    val text: String
)
